app/apollo/resolvers/group.js

```diff
@@ -622,21 +656,23 @@ const groupResolvers = {
       const user = whoIs(me);
 
       try {
-        logger.info({ req_id, user, org_id, uuid, clusters }, `${queryName} validating`);
+        logger.info({req_id, user, org_id, uuid}, `${queryName} validating`);
 
         validateString( 'org_id', org_id );
         validateString( 'uuid', uuid );
         clusters.forEach( value => validateString( 'clusters', value ) );
 
         // validate the group exits in the db first.
         const group = await models.Group.findOne({ org_id: org_id, uuid });
-        if(!group){
-          throw new NotFoundError(context.req.t('group uuid "{{uuid}}" not found', {'uuid':uuid}));
-        }
+        logger.info({req_id, user, org_id, uuid}, `${queryName} validating - found: ${!!group}`);
 
-        await validAuth(me, org_id, ACTIONS.MANAGE, TYPES.GROUP, queryName, context, [group.uuid, group.name]);
+        const identifiers = group ? [uuid, group.name] : [uuid];
+        await validAuth(me, org_id, ACTIONS.MANAGE, TYPES.GROUP, queryName, context, identifiers);
```

Comment:
Does cluster READ auth also needs to be verified? This will require querying clusters from the DB, so the cluster search later on for the pluginContext should be moved earlier, and filtered to the cluster IDs specified.

Reply:
same as above. Fixed but needs review